content/en/blog/2023-06-23-unikraft-gsoc-enhancing-vscode-developer-experience.md

```diff
@@ -0,0 +1,42 @@
++++
+title = "GSoC'23: Enhancing VScode Developer Experience"
+date = "2023-06-23T00:00:00+01:00"
+author = "Md Sahil"
+tags = ["GSoC'23", "VsCode", "IDE Extension"]
++++
+
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-2022-badge.svg" align="right" />
+
+# GSoC'23: Enhancing VScode Developer Experience
+
+## Motivation
+
+The VsCode Extension for unikraft enables developers to quickly and painlessly build unikernels from the VsCode IDE.
+Amongst other features, It allows developers to list and find Unikernel libraries as well as run basic commands.
+In this project, I will upgrade the VSCode extension to use KraftKit, the newly released CLI
+companion tool for unikraft rewritten in GO.
+
+### Summary of Objectives
+
+This project aims to achieve the following outcomes:
+
+- The project requires modifications to the project’s main binary, kraft, To enable JSON output of various commands so that the integration with
+VsCode could be done through a machine-readable interface.
+- And Updating VsCode IDE extension to function properly with new golang based Kraftkit.
+- Additionally, I will enhance the experience, including adding support for additional steps in Unikraft’s build cycle; Packaging unikernels in different formats; Providing a linting.
```

Comment:
```suggestion
* Modify the project's main binary, `kraft`, to enable JSON output of various commands, so that the integration with VS Code could be done through a machine-readable interface.
* Update VS Code IDE extension to function properly with new Go-based KraftKit.
* Additionally, enhance the experience, including adding support for other steps, in Unikraft's build cycle: package unikernels in different formats, provide linting.
```